### PR TITLE
crc32c: fix tests on darwin

### PR DESCRIPTION
**Pull request by r-burns**
Fix test library loading by enabling build-time rpath

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/libraries/crc32c/default.nix [hunk 1]
```diff
@@ -29,6 +29,7 @@ stdenv.mkDerivation rec {
     "-DCRC32C_BUILD_BENCHMARKS=0"
     "-DCRC32C_USE_GLOG=0"
     "-DBUILD_SHARED_LIBS=${if staticOnly then "0" else "1"}"
+    "-DCMAKE_SKIP_BUILD_RPATH=OFF" # for tests
```
Comment (thiagokokada) [inline pkgs/development/libraries/crc32c/default.nix:32]: Why not set this only if `stdenv.isDarwin` :thinking: ?

Comment (r-burns) [reply pkgs/development/libraries/crc32c/default.nix:33]: Sure, may as well

**Review by cpcloud — APPROVED** [review]

**Review by thiagokokada — COMMENTED** [review]
Result of `nixpkgs-review pr 144683` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)

**Review by thiagokokada — APPROVED** [review]
Result of `nixpkgs-review pr 144683` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages marked as broken and skipped:</summary>
  <ul>
    <li>python38Packages.spacy-transformers</li>
    <li>python39Packages.spacy-transformers</li>
  </ul>
</details>
<details>
  <summary>6 packages failed to build:</summary>
  <ul>
    <li>python38Packages.pathy</li>
    <li>python38Packages.spacy</li>
    <li>python38Packages.textacy</li>
    <li>python39Packages.pathy</li>
    <li>python39Packages.spacy</li>
    <li>python39Packages.textacy</li>
  </ul>
</details>
<details>
  <summary>18 packages built:</summary>
  <ul>
    <li>dvc-with-remotes</li>
    <li>python38Packages.gcsfs</li>
    <li>python38Packages.gensim</li>
    <li>python38Packages.google-cloud-automl</li>
    <li>python38Packages.google-cloud-bigquery</li>
    <li>python38Packages.google-cloud-storage</li>
    <li>python38Packages.google-crc32c</li>
    <li>python38Packages.google-resumable-media</li>
    <li>python38Packages.smart-open</li>
    <li>python39Packages.gcsfs</li>
    <li>python39Packages.gensim</li>
    <li>python39Packages.google-cloud-automl</li>
    <li>python39Packages.google-cloud-bigquery</li>
    <li>python39Packages.google-cloud-storage</li>
    <li>python39Packages.google-crc32c</li>
    <li>python39Packages.google-resumable-media</li>
    <li>python39Packages.smart-open</li>
    <li>snakemake</li>
  </ul>
</details>

**Pull request merged**